src/subscription.ts

```diff
@@ -450,17 +487,25 @@ export class Subscription extends EventEmitter {
    *   const exists = data[0];
    * });
    */
-  exists(callback) {
+  exists(): Promise<boolean>;
+  exists(callback: ExistsCallback): void;
+  exists(callback?: ExistsCallback): void|Promise<boolean> {
     this.getMetadata(err => {
       if (!err) {
-        callback(null, true);
+        callback!(null, true);
         return;
       }
-      if (err.code === 5) {
-        callback(null, false);
+      let code = 0;
```

Comment:
This feels pretty messy.  It's ok to use an `any` here, or to create an interface that extends `Error` and adds a `code` property.   Lets minimize behavior changes in pursuit of TypeScript purity :)

Reply:
done!